### DISCUSSION TITLE
how can i build with envd for cmake dockerfile command

### DISCUSSION THREAD

**Discussion by Xiaoaier-Z-L** [tensorchord/envd#1747] [General]
one example dockerfile command: 
RUN CUDAFLAGS="-include stdio.h" cmake \
      -D CMAKE_EXPORT_COMPILE_COMMANDS=1 \
      -D CMAKE_BUILD_TYPE=Release \
      -D ENABLE_FP8=OFF \
      -D CMAKE_INSTALL_PREFIX=/opt/tritonserver \
      -D TRITON_COMMON_REPO_TAG="r${NVIDIA_TRITON_SERVER_VERSION}" \
      -D TRITON_CORE_REPO_TAG="r${NVIDIA_TRITON_SERVER_VERSION}" \
      -D TRITON_BACKEND_REPO_TAG="r${NVIDIA_TRITON_SERVER_VERSION}" \
      .. && \
    cd _deps/repo-ft-src/ && \
    git log | head -n 3 2>&1 | tee /workspace/build/fastertransformer_backend/FT_version.txt && \
    cd /workspace/build/fastertransformer_backend/build && \
    CUDAFLAGS="-include stdio.h" make -O -j"$(grep -c ^processor /proc/cpuinfo)" install && \
    rm /workspace/build/fastertransformer_backend/build/bin/*_example -rf && \
    rm /workspace/build/fastertransformer_backend/build/lib/lib*Backend.so -rf

**Comment by kemingy** [comment]
You can try:

```python
fast_transformer_cmd = """
CUDAFLAGS="-include stdio.h" cmake
-D CMAKE_EXPORT_COMPILE_COMMANDS=1
-D CMAKE_BUILD_TYPE=Release
-D ENABLE_FP8=OFF
-D CMAKE_INSTALL_PREFIX=/opt/tritonserver
-D TRITON_COMMON_REPO_TAG="r${NVIDIA_TRITON_SERVER_VERSION}"
-D TRITON_CORE_REPO_TAG="r${NVIDIA_TRITON_SERVER_VERSION}"
-D TRITON_BACKEND_REPO_TAG="r${NVIDIA_TRITON_SERVER_VERSION}"
.. &&
cd _deps/repo-ft-src/ &&
git log | head -n 3 2>&1 | tee /workspace/build/fastertransformer_backend/FT_version.txt &&
cd /workspace/build/fastertransformer_backend/build &&
CUDAFLAGS="-include stdio.h" make -O -j"$(grep -c ^processor /proc/cpuinfo)" install &&
rm /workspace/build/fastertransformer_backend/build/bin/_example -rf &&
rm /workspace/build/fastertransformer_backend/build/lib/libBackend.so -rf
"""

def build():
    run(commands=[fast_transformer_cmd])
```

**Reply by Xiaoaier-Z-L** [reply]
got, thx.

**Reply by Xiaoaier-Z-L** [reply]
and can i make it as a myself build functions to reuse them by importing it with include?

**Reply by kemingy** [reply]
Of course. You can store it in a git repo and import with `include("github.com/xxx/xxx")` to reuse it.

**Reply by kemingy** [reply]
Refer to:
- https://github.com/tensorchord/envdlib
- https://envd.tensorchord.ai/api/starlark/lib/lang.html

**Reply by Xiaoaier-Z-L** [reply]
Thank you, it's great for me.
